students/ZidanLuo/session09/donor_models.py

```diff
@@ -0,0 +1,193 @@
+#!/usr/bin/env python
+import sys
+import os
+
+class Donor_Collection():
+    def __init__(self):
+        self.donor_dict = {"Fred Jones": Donor("Fred Jones", 500)}
+        self.sorted_dict = {"Fred Jones": Donor("Fred Jones", 500)}
+
+        self.longest_name = "Fred Jones"
+        self.largest_amount = 500
+
+    
+    def copy_constructor(self, sampleDB, longest_name, largest_amount):
```

Comment:
not a great name -- it's not really copying, it's setting with a bunch of data that are in a different format.

If you do want to make a "real" copy constructor, you can define a __copy__ method, and then the Python copy module will use it.